storage: fix errant log message caused by load-based splitting

tbg opened the pull request:
When a load-based split succeeded, the split finder would still be
primed with the same split key for a while; if the split queue picked up
the range again, it would attempt another split at what was already the
range's end key.

Make sure to reset the split finder after a successful load based split.

Closes #34264.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/34291)
<!-- Reviewable:end -->

andreimatei reviewed (approved):
LGTM

good deal

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/34291#-:-LXKx4xu0etXHut_uyX9:b-4up2ui)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andreimatei and @tbg)

___
*[pkg/storage/split/decider_test.go, line 159 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/34291#-LXKwxiu0j5XiCQG-ZKn:-LXKwxiu0j5XiCQG-ZKo:bb5kycn) ([raw file](https://github.com/cockroachdb/cockroach/blob/454669a9d675b219cf97f7820b52891386c699d9/pkg/storage/split/decider_test.go#L159)):*
> ```Go
> 	assert.Equal(t, roachpb.Key(nil), d.MaybeSplitKey(ms(tick)))
> 
> 	for i := 0; i < 1000; i++ {
> ```

I haven't read this diff, nor am I particularly interested in reading it, but I sense that a comment about what we're testing might be welcomed.


<!-- Sent from Reviewable.io -->

tbg committed:
storage: fix errant log message caused by load-based splitting

When a load-based split succeeded, the split finder would still be
primed with the same split key for a while; if the split queue picked up
the range again, it would attempt another split at what was already the
range's end key.

Make sure to reset the split finder after a successful load based split.

Closes #34264.

Release note: None

tbg commented:
Done, TFTR!

bors r=andreimatei
